py/pal2.py

```diff
@@ -202,8 +208,8 @@ def report(self):
         if N > self.best and (N > 12500 or N > self.best+500):
             self.best = len(self)
             self.bestphrase = str(self)
-            print '%5d phrases (%5d words) in %3d seconds' % (
-                self.best, self.bestphrase.count(' ')+1, time.clock() - self.starttime)
+            print('%5d phrases (%5d words) in %3d seconds' % (
+                self.best, self.bestphrase.count(' ')+1, time.clock() - self.starttime))
             assert is_panama(self.bestphrase)
             f = open('pallog%d.txt' % (id(self) % 10000), 'w')
             f.write(self.bestphrase + '\n')
```

Comment:
```suggestion
            with open('pallog%d.txt' % (id(self) % 10000), 'w') as f:
                f.write(self.bestphrase + '\n')
```

Reply:
I always use the __with open() as__ syntax in my own code but this is not my own code.

Given the genius of this code's original author, my preference is to make only those minimal changes required to achieve Python 3 compatibility.

Reply:
Understandable. So keep unless @norvig thinks it should be updated.